libavpipe/src/avpipe_mux.c

```diff
@@ -379,20 +378,20 @@ get_next_packet(
         if (pkts[index].pts == pkt->pts)
             goto read_frame_again;
         xctx->is_pkt_valid[index] = 1;
-        if (pkt->pts >= 0) {
+        if (pkt->dts >= 0) {
             if (index == 0) {
-                if (pkt->pts > in_mux_ctx->last_video_pts)
-                    in_mux_ctx->last_video_pts = pkt->pts;
+                if (pkt->dts > in_mux_ctx->last_video_dts)
+                    in_mux_ctx->last_video_dts = pkt->dts;
                 else {
-                    pkt->pts += in_mux_ctx->last_video_pts;
-                    pkt->dts += in_mux_ctx->last_video_pts;
+                    pkt->pts += in_mux_ctx->last_video_dts;
+                    pkt->dts += in_mux_ctx->last_video_dts;
                 }
             } else if (index <= in_mux_ctx->last_audio_index) {
-                if (pkt->pts > in_mux_ctx->last_audio_pts)
-                    in_mux_ctx->last_audio_pts = pkt->pts;
+                if (pkt->dts > in_mux_ctx->last_audio_dts)
+                    in_mux_ctx->last_audio_dts = pkt->dts;
                 else {
-                    pkt->pts += in_mux_ctx->last_audio_pts;
-                    pkt->dts += in_mux_ctx->last_audio_pts;
+                    pkt->pts += in_mux_ctx->last_audio_dts;
+                    pkt->dts += in_mux_ctx->last_audio_dts;
```

Comment:
Ah, it took a while to figure out, but I think this code is to handle each mezzanine part starting from PTS 0. I wonder if it might be better just to handle PTS and DTS adjustments separately, rather than setting PTS to DTS or vice versa. Also, it seems like whenever the input DTS resets to 0, we'd get a duplicate DTS, e.g. if `last_video_dts` is 100, and the next DTS is 0, we get a packet with DTS 100 again.